core/dbt/task/run.py

```diff
@@ -487,56 +341,191 @@ def execute(self, model, manifest):
             )
 
         hook_ctx = self.adapter.pre_model_hook(context_config)
-        if (
-            model.config.materialized == "incremental"
-            and model.config.incremental_strategy == "microbatch"
-            and manifest.use_microbatch_batches(project_name=self.config.project_name)
-        ):
-            return self._execute_microbatch_model(
-                hook_ctx, context_config, model, manifest, context, materialization_macro
-            )
+
+        return self._execute_model(
+            hook_ctx, context_config, model, manifest, context, materialization_macro
+        )
+
+
+class MicrobatchModelRunner(ModelRunner):
+    batch_idx: Optional[int] = None
+    batches: Dict[int, BatchType] = field(default_factory=dict)
+    relation_exists: bool = False
+
+    def set_batch_idx(self, batch_idx: int) -> None:
+        self.batch_idx = batch_idx
+
+    def set_relation_exists(self, relation_exists: bool) -> None:
+        self.relation_exists = relation_exists
+
+    def set_batches(self, batches: Dict[int, BatchType]) -> None:
+        self.batches = batches
+
+    def describe_node(self) -> str:
+        return f"{self.node.language} microbatch model {self.get_node_representation()}"
+
+    def describe_batch(self, batch_start: Optional[datetime]) -> str:
+        # Only visualize date if batch_start year/month/day
+        formatted_batch_start = MicrobatchBuilder.format_batch_start(
+            batch_start, self.node.config.batch_size
+        )
+        import threading
+
+        return f"batch {formatted_batch_start} of {self.get_node_representation()} on {threading.current_thread().name}"
```

Comment:
let's remove the `on threading.current_thread().name` here, it was just for development

Reply:
Hmm, this is probably going to clutter up the logs for users. I wonder if we get any information about utilized threads in the debug logs we can leverage instead for the purposes of testing

Reply:
What do you think about extending the debug `NodeStart` with a `thread_name` and passing the thread_name through GraphRunnableTask.call_runner (where NodeStart is currently fired)

Reply:
🤔 Let's try it! We're still waiting on the approval of the dbt-adapters PR anyway